src/coreclr/jit/lsra.cpp

```diff
@@ -3856,9 +3856,37 @@ void LinearScan::processKills(RefPosition* killRefPosition)
     RefPosition* nextKill = killRefPosition->nextRefPosition;
 
     regMaskTP killedRegs = killRefPosition->getKilledRegisters();
-    while (killedRegs.IsNonEmpty())
+
+    freeKilledRegs(killRefPosition, killedRegs.getLow(), nextKill, REG_LOW_BASE);
+
+#ifdef HAS_MORE_THAN_64_REGISTERS
+    freeKilledRegs(killRefPosition, killedRegs.getHigh(), nextKill, REG_HIGH_BASE);
+#endif
+
+    regsBusyUntilKill &= ~killRefPosition->getKilledRegisters();
+    INDEBUG(dumpLsraAllocationEvent(LSRA_EVENT_KILL_REGS, nullptr, REG_NA, nullptr, NONE,
+                                    killRefPosition->getKilledRegisters()));
+}
+
+//------------------------------------------------------------------------
+// freeKilledRegs: Handle  registers that are being killed.
+//
+// Arguments:
+//   killRefPosition - The RefPosition for the kill
+//   killedRegs - Registers to kill
+//   nextKill - The RefPosition for next kill
+//   regBase - `0` or `64` based on the `killedRegs` being processed
+//
+void LinearScan::freeKilledRegs(RefPosition* killRefPosition,
+                                regMaskSmall killedRegs,
+                                RefPosition* nextKill,
+                                int          regBase)
+{
+
+    while (killedRegs != RBM_NONE)
     {
-        regNumber  killedReg        = genFirstRegNumFromMaskAndToggle(killedRegs);
+        regNumber killedReg = (regNumber)(BitOperations::BitScanForward(killedRegs) + regBase);
```

Comment:
consider having an equivalent method of `genFirstRegNumFromMaskAndToggle()` that takes `SingleTypeRegSet` as parameter and operate on it. You can also pass `lowBase` and `highBase`, if required.

Reply:
You are right. I should not have added `regBase` where I did either. The following xor is slightly incorrect.

Should be

`
regNumber killedReg = (regNumber)(genFirstRegNumFromMaskAndToggle(killedRegs) + regBase);
  RegRecord* regRecord        = getRegisterRecord(killedReg);
`

Reply:
Made this change as well. I'm done making changes now